### ISSUE TITLE
[BUG] need explanation about tld filter

### ISSUE THREAD

**Issue by kasak87** [rspamd/rspamd#4275]
Greetings!
I have this simple filter in local.d/multimap.conf:
`PHISHING_DOMAIN {
 type = "url";
 filter = "tld";
 map = "$LOCAL_CONFDIR/local.d/url.map"; 
}
`
in this filter i'm including urls that often repeat in phishing letters. 
it works flawless, but i have a problem with one url: b24-5ba83q.bitrix24.site
I added it to url.map but it does not work :( 
only after i cut it to "bitrix24.site" it works!

in documentation to rspamd i see:
_tld matches eSLD (effective second level domain - a second-level domain or something that’s effectively so like example.com or example.za.org)_
So, b24-5ba83q it third level domain and that's why it is not working? Or this is a bug? 

Rspamd version 3.2 in openbsd 7.1 packages

**Comment by kasak87** [comment]
oh, i immediately have an addition! 
www.zimbra2022.c1.biz 
this is also not working until i cut it to 
c1.biz 
so i think this is not a bug, this is my misunderstanding of docs :( 
is there any way i can include more than second layer domain?

**Comment by frzqwerty** [comment]
Is there any improvement or fix about that issue, so far?

**Comment by moisseev** [comment]
> _tld matches eSLD (effective second level domain - a second-level domain or something that’s effectively so like example.com or example.za.org)_
> So, b24-5ba83q it third level domain and that's why it is not working?

Exactly. The `tld` filter means Rspamd should extract eSLD part (`bitrix24.site`) from the URL and compare it with the map entries. So, it works as expected.

Probably you should use `regexp` filter (and regexp map) instead.

**Comment by vstakhov** [comment]
Not a bug.
